perceval/backends/core/github.py

```diff
@@ -484,17 +484,17 @@ def __init__(self, owner, repository, token, base_url=None,
 
         self.min_rate_to_sleep = min(min_rate_to_sleep, MAX_RATE_LIMIT)
 
-    def get_rate_limit(self):
-        """Get rate limit"""
+    def init_rate_limit(self):
+        """initialize rate_limit and rate_limit_reset_ts """
 
         url = urijoin(self.api_url, "rate_limit")
-        response = requests.get(url, headers=self.__build_headers())
-        response.raise_for_status()
-
-        return response
-
-    def init_rate_limit(self, response):
-        """initialize rate_limit and rate_limit_reset_ts """
+        try:
+            response = requests.get(url, headers=self.__build_headers())
+            response.raise_for_status()
+        except requests.exceptions.HTTPError:
+            self.rate_limit = None
+            self.rate_limit_reset_ts = None
+            return
```

Comment:
Instead of calling return here, it's better to remove it and put the next two lines in an `else` block, like this:

```
self.rate_limit = None
self.rate_limit_reset_ts = None

try:
     [...]
except requests.exceptions.HTTPError as e:
    # When rate limit is not available in GitHub Enterprise instances,
    # a HTTP 404 error is received
    if e.status_code != 404:
        raise e
else:
     self.rate_limit = int(response.headers[self.RATE_LIMIT_HEADER])
     self.rate_limit_reset_ts = int(response.headers[self.RATE_LIMIT_RESET_HEADER])
```